packages/amplify-provider-awscloudformation/lib/attach-backend.js

```diff
@@ -0,0 +1,299 @@
+  } else if (backendEnvs.length === 1) {
+    context.print.info(`Found one backend environment '${backendEnvs[0].environmentName}'`);
+    const confirmMigrateMessage = 'Do you want to choose it?';
+    if (await context.prompt.confirm(confirmMigrateMessage)) {
+      return backendEnvs[0];
+    }
+  } else {
+    const errorMessage = `Found no currently existing backend environment in the amplify project: ${amplifyApp.name}.`;
+    context.print.error(errorMessage);
+    throw new Error(errorMessage);
+  }
+}
+
+async function downloadBackend(context, backendEnv, awsConfig) {
+  if (!backendEnv) {
+    return;
+  }
+  const amplifyDirPath = context.amplify.pathManager.getAmplifyDirPath(process.cwd());
+  const tempDirPath = path.join(amplifyDirPath, 'temp');
+  const currentCloudBackendDir = context.amplify.pathManager.getCurrentCloudBackendDirPath(process.cwd());
+  const backendDir = context.amplify.pathManager.getBackendDirPath(process.cwd());
+  const zipFileName = constants.S3BackendZipFileName;
+
+  const s3Client = getConfiguredS3Client(awsConfig);
+  const deploymentBucketName = backendEnv.deploymentArtifacts;
+
+  const params = {
+    Key: zipFileName,
+    Bucket: deploymentBucketName,
+  };
+
+  const zipObject = await s3Client.getObject(params).promise();
+  const buff = Buffer.from(zipObject.Body);
+
+  fs.ensureDirSync(tempDirPath);
+  const tempFilePath = path.join(tempDirPath, zipFileName);
+  fs.writeFileSync(tempFilePath, buff);
+
+  const unzippedDirPath = path.join(tempDirPath, path.basename(zipFileName, '.zip'));
+
+  await new Promise((res, rej) => {
+    extract(tempFilePath, { dir: unzippedDirPath }, err => {
+      if (err) {
+        rej(err);
+      }
+      res(unzippedDirPath);
+    });
+  });
+
+  fs.copySync(unzippedDirPath, currentCloudBackendDir);
+  fs.copySync(unzippedDirPath, backendDir);
+  fs.removeSync(tempDirPath);
+}
+
+async function getAwsConfig(context) {
+  const { awsConfigInfo } = context.exeInfo;
+  const httpProxy = process.env.HTTP_PROXY || process.env.HTTPS_PROXY;
+
+  let awsConfig;
+  if (awsConfigInfo.configLevel === 'project') {
+    if (awsConfigInfo.config.useProfile) {
+      awsConfig = await systemConfigManager.getProfiledAwsConfig(context, awsConfigInfo.config.profileName);
+    } else {
+      awsConfig = {
+        accessKeyId: awsConfigInfo.config.accessKeyId,
+        secretAccessKey: awsConfigInfo.config.secretAccessKey,
+        region: awsConfigInfo.config.region,
+      };
+    }
+  }
+
+  if (httpProxy) {
+    awsConfig = {
+      ...awsConfig,
+      httpOptions: { agent: proxyAgent(httpProxy) },
+    };
+  }
+
+  return awsConfig;
+}
+
+function getConfiguredS3Client(awsConfig) {
+  const httpProxy = process.env.HTTP_PROXY || process.env.HTTPS_PROXY;
+
+  if (httpProxy) {
+    aws.config.update({
+      httpOptions: {
+        agent: proxyAgent(httpProxy),
+      },
+    });
+  }
+
+  return new aws.S3({ ...awsConfig });
+}
+
+module.exports = {
+  run,
+};
```

Comment:
I can not use the S3 client directly because in its upload files method, it's expecting a project that's already setup by the Amplify CLI and has the teamProviderInfo and LocalEnvInfo etc. In the attach backend workflow, those items are not there and are to be setup by the workflow. 
We need refactor the S3 client in the aws-utils to make it more flexible.